tests/queries/test_qs_combinators.py

```diff
@@ -119,6 +119,17 @@ def test_union_with_values(self):
         reserved_name = qs1.union(qs1).values_list('name', 'order', 'id').get()
         self.assertEqual(reserved_name[:2], ('a', 2))
 
+    def test_union_with_two_annotated_values_list(self):
+        qs1 = Number.objects.filter(num=1).annotate(
+            count=Value(0, IntegerField()),
+        ).values_list('id', 'count')
+        qs2 = Number.objects.filter(num=0).values('num').annotate(
+            id=F('pk'),
```

Comment:
Relying on pk might be problematic since we shouldn't assume the values that the database might assign.

Reply:
Agreed. Maybe:
```python
def test_union_with_two_annotated_values_list(self):
    qs1 = Number.objects.filter(num=1).annotate(
        count=Value(0, IntegerField()),
    ).values_list('num', 'count')
    qs2 = Number.objects.filter(num=2).values('pk').annotate(
        count=F('num'),
        num=Value(1, IntegerField()),
    ).values_list('num', 'count')
    self.assertCountEqual(qs1.union(qs2), [(1, 0), (2, 1)])
```